src/components/TodoHeader.tsx

```diff
@@ -0,0 +1,64 @@
+import { Dispatch, SetStateAction, useEffect, useRef, useState } from 'react';
+import { Error } from '../types/Error';
+import { Todo } from '../types/Todo';
+
+type Props = {
+  onAddTodo: (value: string) => Promise<void>;
+  setErrorMessage: Dispatch<SetStateAction<Error | null>>;
+  isInputDisabled: boolean;
+  todos: Todo[];
+};
+export const TodoHeader: React.FC<Props> = props => {
+  const { onAddTodo, setErrorMessage, isInputDisabled, todos } = props;
+
+  const [inputValue, setInputValue] = useState('');
+
+  const inputRef = useRef<HTMLInputElement>(null);
+
+  const onSubmit = async (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+    if (inputValue.trim() === '') {
+      setErrorMessage(Error.EmptyTitle);
+
+      return;
+    }
+
+    try {
+      await onAddTodo(inputValue.trim());
+      setInputValue('');
+    } catch (err) {}
+  };
+
+  useEffect(() => {
+    if (inputRef.current) {
```

Comment:
you don't have a check for presence TempTodo for focusing
there is checked in tests